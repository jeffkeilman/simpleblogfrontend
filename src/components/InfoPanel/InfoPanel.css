.Description {
  display: inline-block;
  height: auto;
  padding-left: 15px;
  vertical-align: top;
}

.Description p {
  font-size: 2em;
  font-weight: 220;
}

.Description span {
  font-weight: 500;
}

.InfoPanel {
  display: inline-block;
  vertical-align: top;
  width: 70%;
  line-height: 1;
}

.LinkParagraph {
  margin-bottom: 10px;
}

.LinkedInLink {
  padding-left: 15px;
}

@media screen and (max-width: 1175px) {
  .InfoPanel  {
    width: 100%
  }
}
