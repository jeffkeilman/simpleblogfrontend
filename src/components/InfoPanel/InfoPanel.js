import './InfoPanel.css'
import React from 'react'
import GitHubLogo from './assets/GitHubMarkLogo.png'
import LinkedInLogo from './assets/LinkedInLogo.png'

class Description extends React.Component {
  render () {
    return (
      <section className='InfoPanel'>
        <section className='Description'>
          <p>Hello. My name is <span>{this.props.name}</span>.</p>
          <p>I am a <span>software engineer</span> with a passion for <span>social justice</span>.</p>
          <p className='LinkParagraph'>Here are some of my <span>other links</span>:</p>
          {!!this.props.githuburl &&
            // if githuburl exists, render GitHub logo link
            <a href={this.props.githuburl} target='_blank' rel='noopener noreferrer'>
              <img
                src={GitHubLogo}
                alt='GitHub link'
                className='GitHubLink'
              />
            </a>}
          {!!this.props.linkedinurl &&
            // if linkedinurl exists, render LinkedIn logo link
            <a href={this.props.linkedinurl} target='_blank' rel='noopener noreferrer'>
              <img
                src={LinkedInLogo}
                alt='LinkedIn link'
                className='LinkedInLink'
              />
            </a>}
          <p>And here is my <span>blog</span>.</p>
        </section>
      </section>
    )
  }
}

export default Description
