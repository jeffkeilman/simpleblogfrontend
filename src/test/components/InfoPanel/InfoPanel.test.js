import React from 'react'
import renderer from 'react-test-renderer'
import InfoPanel from '../../../components/InfoPanel/InfoPanel'
import { render } from '@testing-library/react'

describe('InfoPanel', () => {
  // snapshot test
  it('renders with links correctly', () => {
    const ip = renderer.create(<InfoPanel name='foobar' linkedinurl='foo' githuburl='bar' />).toJSON()
    expect(ip).toMatchSnapshot()
  })

  // snapshot test
  it('renders without links correctly', () => {
    const ip = renderer.create(<InfoPanel name='foobar' />).toJSON()
    expect(ip).toMatchSnapshot()
  })

  it('renders the description section', () => {
    const { container } = render(<InfoPanel name='foobar' linkedinurl='foo' githuburl='bar' />)
    expect(container.querySelector('.Description')).toBeInTheDocument()
  })

  describe('with links', () => {
    // TODO: I can't seem to do all of these with one render. Would like that
    it('renders the link paragraph', () => {
      const { container } = render(<InfoPanel name='foobar' linkedinurl='foo' githuburl='bar' />)
      expect(container.querySelector('.LinkParagraph')).toBeInTheDocument()
    })

    it('renders the GitHub link', () => {
      const { queryByAltText } = render(<InfoPanel name='foobar' linkedinurl='foo' githuburl='bar' />)
      const githubIcon = queryByAltText(/GitHub link/i)
      const a = githubIcon.closest('a')

      expect(a).toHaveAttribute('href', 'bar')
      expect(githubIcon).toBeInTheDocument()
    })

    it('renders the LinkedIn link', () => {
      const { queryByAltText } = render(<InfoPanel name='foobar' linkedinurl='foo' githuburl='bar' />)
      const linkedinIcon = queryByAltText(/LinkedIn link/i)
      const a = linkedinIcon.closest('a')

      expect(a).toHaveAttribute('href', 'foo')
      expect(linkedinIcon).toBeInTheDocument()
    })
  })

  describe('without links', () => {
    // TODO: I can't seem to do all of these with one render. Would like that
    it('does not render the link paragraph', () => {
      const { container } = render(<InfoPanel name='foobar' />)
      expect(container.querySelector('.LinkParagraph')).toBeFalsy()
    })

    it('does not render any links', () => {
      const { container } = render(<InfoPanel name='foobar' />)
      expect(container.querySelector('a')).toBeFalsy()
    })
  })
})
