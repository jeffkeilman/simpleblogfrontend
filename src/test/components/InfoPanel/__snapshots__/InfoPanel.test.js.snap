// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoPanel renders with links correctly 1`] = `
<section
  className="InfoPanel"
>
  <section
    className="Description"
  >
    <p>
      Hello. My name is 
      <span>
        foobar
      </span>
      .
    </p>
    <p>
      I am a 
      <span>
        software engineer
      </span>
       with a passion for 
      <span>
        social justice
      </span>
      .
    </p>
    <p
      className="LinkParagraph"
    >
      Here are some of my 
      <span>
        other links
      </span>
      :
    </p>
    <a
      href="bar"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="GitHub link"
        className="GitHubLink"
        src="GitHubMarkLogo.png"
      />
    </a>
    <a
      href="foo"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="LinkedIn link"
        className="LinkedInLink"
        src="LinkedInLogo.png"
      />
    </a>
    <p>
      And here is my 
      <span>
        blog
      </span>
      .
    </p>
  </section>
</section>
`;

exports[`InfoPanel renders without links correctly 1`] = `
<section
  className="InfoPanel"
>
  <section
    className="Description"
  >
    <p>
      Hello. My name is 
      <span>
        foobar
      </span>
      .
    </p>
    <p>
      I am a 
      <span>
        software engineer
      </span>
       with a passion for 
      <span>
        social justice
      </span>
      .
    </p>
    <p>
      And here is my 
      <span>
        blog
      </span>
      .
    </p>
  </section>
</section>
`;
